apps/website/src/components/forms/EntryForm.tsx

```diff
@@ -55,92 +58,182 @@ export const EntryForm: React.FC<{
-          Street address: {existingEntry.mailingAddress?.addressLine1}
-          <br />
-          Second address line:{" "}
-          {existingEntry.mailingAddress?.addressLine2 || "-"}
-          <br />
-          City: {existingEntry.mailingAddress?.city}
-          <br />
-          State / Province / Region: {existingEntry.mailingAddress?.state}
-          <br />
-          Postal code/ZIP: {existingEntry.mailingAddress?.postalCode}
-          <br />
-          Country:{" "}
-          {existingEntry.mailingAddress?.country &&
-            getCountryName(existingEntry.mailingAddress?.country)}
-        </p>
-      </>
-    );
-  }
+  // Map submitted form data, or existing entry data, to a consistent format
+  const entry = enterForm.isSuccess
+    ? enterForm.variables
+    : existingEntry && {
+        email: existingEntry.email,
+        givenName: existingEntry.givenName,
+        familyName: existingEntry.familyName,
+        addressLine1: existingEntry.mailingAddress?.addressLine1,
+        addressLine2: existingEntry.mailingAddress?.addressLine2,
+        country: existingEntry.mailingAddress?.country,
+        state: existingEntry.mailingAddress?.state,
+        city: existingEntry.mailingAddress?.city,
+        postalCode: existingEntry.mailingAddress?.postalCode,
+      };
 
   return (
-    <form onSubmit={handleSubmit}>
-      {config.intro && <Markdown content={config.intro} />}
-
-      {enterForm.error && (
-        <MessageBox variant="failure">
-          Error: {enterForm.error.message}
-        </MessageBox>
+    <>
+      <Section dark>
+        <header>
+          <Heading>{form.label}</Heading>
+        </header>
+
+        {config.intro && <Markdown content={config.intro} />}
+      </Section>
+
+      {/* Handle users that aren't logged in */}
+      {!session?.user?.id && (
+        <Section containerClassName="max-w-lg">
+          <MessageBox>
+            <p className="mb-4">
+              You need to be logged in with Twitch to enter.
+            </p>
+
+            <LoginWithTwitchButton />
+          </MessageBox>
+        </Section>
       )}
 
-      {config.checks && (
+      {/* Handle users that are logged in, but have already submitted */}
+      {session?.user?.id && entry && (
         <>
-          <Headline>Steps to enter</Headline>
-          <GiveawayChecks />
+          <Section containerClassName="max-w-lg">
+            <MessageBox variant="success">
+              <Heading level={2} className="mx-2 mb-2 text-2xl">
+                {enterForm.isSuccess
+                  ? "Your entry has been submitted!"
+                  : "You've already entered!"}
+              </Heading>
+
+              <p className="m-2">Check the details you submitted below.</p>
+            </MessageBox>
+          </Section>
+
+          <Section dark>
+            <Promos />
+          </Section>
+
+          <Section containerClassName="max-w-xl">
+            <Heading level={2} className="mb-6 text-2xl">
```

Comment:
I do not really the promo here and i don't think its that necessary.

```suggestion
            <Heading level={2} className="mb-6 mt-12 text-2xl">
```

You could also make the section extra dark (`className="bg-alveus-green-green-900"`) and move it below the entry, but then we already have the socials section there.

Reply:
Yah, that's why I'd placed it in the middle here, as it felt a bit weird to have an updates promo right above another mention of updates again.

Reply:
Can drop it from here though, I think having it on the form with checks should be enough